vignettes/creating_linters.Rmd

```diff
@@ -11,113 +11,186 @@ vignette: >
-          column_number = parsed$col1,
-          type = "style",
-          message = "Use <-, not =, for assignment.",
-          line = source_file$lines[parsed$line1]
-        )
-      }
+pipe_call_linter <- function() {
+  Linter(function(source_expression) {
+    if (!is_lint_level(source_expression, "expression", require_xml = TRUE)) {
+      return(list())
+    }
+
+    xml <- source_expression$xml_parsed_content
+
+    xpath <- "//expr[preceding-sibling::SPECIAL[text() = '%>%'] and *[1][self::SYMBOL]]"
+    bad_expr <- xml2::xml_find_all(xml, xpath)
+
+    xml_nodes_to_lint(
+      bad_expr,
+      source_expression = source_expression,
+      lint_message = "Use explicit calls in magrittr pipes, i.e., `a %>% foo` should be `a %>% foo()`.",
+      type = "warning"
     )
   })
 }
 ```
 
-Let us walk through the parts of the linter individually.
+Let's walk through the parts of the linter individually.
 
 ## Writing the linter ##
+
 ```r
-#' @describeIn linters  Check that '<-' is always used for assignment.
+#' Pipe call linter
+#'
+#' Force explicit calls in magrittr pipes, e.g.,
+#' `1:3 %>% sum()` instead of `1:3 %>% sum`.
+```
+
+Describe the linter, giving it a title and briefly covering the usages that are discouraged when the linter is active.
+
+```r
+#' @evalRd rd_tags("pipe_call_linter")
+#' @seealso [linters] for a complete list of linters available in lintr.
 #' @export
 ```
-The first two lines add the linter to the `linters` documentation and export it
-for use outside the package.
+
+These lines (1) generate a Tags section in the documentation for the linter; (2) link to the full table of available linters; and (3) mark the function for export. The most unfamiliar here is probably (1); this uses a helper function inside lintr to generate custom Rd styling. See R/linter_tags.R.
 
 ```r
-assignment_linter <- function() {
+pipe_call_linter <- function() {
 ```
+
 Next, we define the name of the new linter.  The convention is to suffix all
 linter names with `_linter`. All `_linter` functions are getters that return a
-closure that will do the actual linting. We could define additional parameters
-that are useful for the linter in this function declaration, but
-`assignment_linter` requires no additional arguments.
+closure that will do the actual linting (i.e., they're function factories). We could define additional parameters
+that are useful for the linter in this function declaration (see, e.g. `assignment_linter`), but
+`pipe_call_linter` requires no additional arguments.
 
 ```r
-Linter(function(source_file) {
+Linter(function(source_expression) {
 ```
-This is the closure. It will be called on the `source_file` variable that
+
+This is the closure. It will be called on the `source_expression` variable that
 contains the top level expressions in the file to be linted. The call to
-`Linter()` simply gives this closure the class 'linter'.
+`Linter()` gives this closure the class 'linter' and guesses its name.
```

Comment:
Not 100% sure if the "and guesses its name" is clear.

Reply:
pushed into parenthetical with doc reference